### PR TITLE
Convert amplicon16S_analysis workflow (QIIME) from single-sample to multi-sample

### PR DESCRIPTION
**Pull request by golu099**
Updating `tasks_16S_amplicon` to consider multiple files for analysis. Curious to know if theres any suggestion to optimization for this run?

### CODE REVIEW

**Comment by golu099** [comment]
@dpark01 @tomkinsc do you mind reviewing this to be merged onto main branch?

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 1]
```diff
@@ -5,47 +5,53 @@ task qiime_import_from_bam {
...
-        File    reads_bam
-        String  sample_name
-        Int     memory_mb = 2000
-        Int     cpu = 1
-        Int     disk_size_gb = ceil(2*size(reads_bam, "GiB")) + 5
+        Array[File] reads_bam
+        Int     memory_mb = 7000
+        Int     cpu = 5
+        Int     disk_size_gb = ceil(2*20) + 5
         String  docker     = "quay.io/broadinstitute/qiime2:conda"
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:12]: Can you do one more thing and go through the whole `tasks_16S_amplicon.wdl` file and replace all these docker strings with the simpler docker string below (without the `:conda`)?

```suggestion
        String  docker     = "quay.io/broadinstitute/qiime2" 
```

**Review by dpark01 — CHANGES_REQUESTED** [review]

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 2]
```diff
@@ -5,47 +5,53 @@ task qiime_import_from_bam {
...
-        Int     disk_size_gb = ceil(2*size(reads_bam, "GiB")) + 5
+        Array[File] reads_bam
+        Int     memory_mb = 7000
+        Int     cpu = 5
+        Int     disk_size_gb = ceil(2*20) + 5
         String  docker     = "quay.io/broadinstitute/qiime2:conda" 
     }
     parameter_meta {
-        reads_bam: {description: "Input BAM file"}
+        reads_bam: {description: "Input BAM files"}
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:15]: Can you replace this description/documentation with something a bit more descriptive of what is in the file? (e.g. "unaligned reads")

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 3]
```diff
@@ -5,47 +5,53 @@ task qiime_import_from_bam {
...
-        #\t= tabs next value 
-        echo -e "$NEWSAMPLENAME\t$(pwd)/R1.fastq.gz\t$(pwd)/R2.fastq.gz" >> manifest.tsv
-        
+        for bam in ~{sep=' ' reads_bam}; do
+            #making new bash variable | regex: (_) -> (-)
+            NEWSAMPLENAME=$(basename $bam .bam  | perl -lape 's/[_]/-/g')
+            echo $NEWSAMPLENAME
+            samtools fastq -1 $NEWSAMPLENAME.R1.fastq.gz -2 $NEWSAMPLENAME.R2.fastq.gz -0 /dev/null $bam
+            #All names added to one giant file 
+            #up to here works...not reading the manifest tsv for some reason
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:30]: Does it work now? (reading the manifest tsv)
```suggestion
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 4]
```diff
@@ -63,15 +69,13 @@ task trim_reads {
...
-        
-        String  qza_basename = basename(reads_qza, '.qza')
         #Boolean not_default = false
         String  forward_adapter         = "CTGCTGCCTCCCGTAGGAGT"
         String  reverse_adapter         = "AGAGTTTGATCCTGGCTCAG"
         Int    min_length              = 1
         Boolean keep_untrimmed_reads   = false
         Int     memory_mb = 2000
-        Int     cpu = 1
+        Int     cpu = 3
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:78]: Let's just make it an even number--I haven't seen most platforms provide 3 CPUs for a VM
```suggestion
        Int     cpu = 4
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 5]
```diff
@@ -63,15 +69,13 @@ task trim_reads {
...
         Int     disk_size_gb = ceil(2*size(reads_qza, "GiB")) + 5
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:79]:
```suggestion
        Int     disk_size_gb = ceil(2*size(reads_qza, "GiB")) + 50
```

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 6]
```diff
@@ -115,7 +119,6 @@ task join_paired_ends {
     input {
         #Input File: Merge paired reads
         File    trimmed_reads_qza
-        String  reads_basename = basename(trimmed_reads_qza, '.qza')
         Int     memory_mb = 2000
         Int     cpu = 1
         Int     disk_size_gb = ceil(2*size(trimmed_reads_qza, "GiB")) + 5
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:124]:
```suggestion
        Int     disk_size_gb = ceil(2*size(trimmed_reads_qza, "GiB")) + 50
```

**Review by dpark01 — CHANGES_REQUESTED** [review]
Small comments and suggestions

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl (hunk 1 above)
Comment (golu099) [reply pipes/WDL/tasks/tasks_16S_amplicon.wdl:12]: Addressed in VS

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl (hunk 3 above)
Comment (golu099) [reply pipes/WDL/tasks/tasks_16S_amplicon.wdl:30]: The results do show the array files being picked up, so it should be.

Comment (dpark01) [reply pipes/WDL/tasks/tasks_16S_amplicon.wdl:30]: Ok, so can you accept the suggested change above?

Comment (golu099) [reply pipes/WDL/tasks/tasks_16S_amplicon.wdl]: Yes, accepted!

File: pipes/WDL/tasks/tasks_16S_amplicon.wdl [hunk 7]
```diff
@@ -9,10 +9,10 @@ task qiime_import_from_bam {
         Int     memory_mb = 7000
         Int     cpu = 5
         Int     disk_size_gb = ceil(2*20) + 5
-        String  docker     = "quay.io/broadinstitute/qiime2:conda" 
+        String  docker     = "quay.io/broadinstitute/qiime2" 
     }
     parameter_meta {
-        reads_bam: {description: "Input BAM files"}
+        reads_bam: {description: "Unaligned reads parsed into FASTQ format and then through a .qza format for QIIME requirements."}
```
Comment (dpark01) [inline pipes/WDL/tasks/tasks_16S_amplicon.wdl:15]:
```suggestion
        reads_bam: {description: "Unaligned reads in BAM format, one sample per BAM file."}
        reads_qza: {description: "All unaligned reads in a single QZA (QIIME) file"}
```

**Review by dpark01 — CHANGES_REQUESTED** [review]

**Review by dpark01 — APPROVED** [review]

**Pull request merged**